tests/pytorch/test_fused_optimizer.py

```diff
@@ -792,6 +792,34 @@ def test_native(self):
 
             self.model_.load_state_dict(copy.deepcopy(self.model.state_dict()))
 
+    def test_optimizer_get_moments_and_variance(self):
+        params_ = [p for p in self.model_.parameters() if p.requires_grad]
+        optimizer_ = te.optimizers.FusedAdam(params_, lr=self.lr, capturable=False)
+
+        for i in range(100):
+            x = torch.rand([32, 1, 28, 28]).cuda().to(memory_format=torch.channels_last)
+            x_ = x.clone()
+            gt = torch.rand([32, 10]).cuda()
+            gt_ = gt.clone()
+
+            # DUT
+            y = self.model_(x)
+            loss_ = ((gt_ - y) ** 2).mean()
+
+            loss_.backward()
+            optimizer_.step()
+
+            # Init for next iteration
+            optimizer_.zero_grad()
+        for param in [p for p in self.model_.parameters() if p.requires_grad]:
+            first_moment = optimizer_.get_unscaled_state(param, "exp_avg")
+            second_moment = optimizer_.get_unscaled_state(param, "exp_avg_sq")
```

Comment:
We could make this test more robust by manually computing `exp_avg` and `exp_avg_sq` outside of the optimizer. There's some complicated dtype-specific logic within `get_unscaled_state`, so I don't really trust it.